endpoint.go

```diff
@@ -61,6 +61,7 @@ type endpoint struct {
 	generic       map[string]interface{}
 	joinLeaveDone chan struct{}
 	prefAddress   net.IP
+	prefAddressV6 net.IP
```

Comment:
I think we should treat it as a multiple ip-address requirement that @jc-m raised.
That will cover both ipv4 and ipv6 cases. WDYT ?